app/app.ts

```diff
@@ -422,10 +424,26 @@ function setupIpc(localSettings: LocalSettings, scrSettings: ScrSettings) {
     mapStore.downloadMap(mapHash, mapFormat, mapUrl),
   )
 
-  ipcMain.handle('replayParseHeader', async (event, replayPath) => {
-    return new Promise(async resolve => {
+  ipcMain.handle('replayParseMetadata', async (event, replayPath) => {
+    return new Promise(async (resolve, reject) => {
       const parser = new ReplayParser()
-      parser.on('replayHeader', header => resolve(header))
+      let headerData: ReplayHeader
+      parser.on('replayHeader', header => {
+        headerData = header
+      })
+
+      let shieldBatteryData: ReplayShieldBatteryData
```

Comment:
this can also be `undefined` now and should be included in this type (and also in whatever type you're resolving to here, if it's not already)